DigitalOffice/Services/UserService/Commands/LoginUserCommand.cs

```diff
@@ -9,43 +9,37 @@
 using UserService.Database;
 using UserService.Database.Entities;
 using UserService.Models;
-using UserService.Validators;
-using Microsoft.AspNetCore.Cryptography.KeyDerivation;
 using System.Security.Cryptography;
 using System.Text;
 using System.Net;
 using System.Web;
-using UserService.Controllers;
+using UserService.Repositories.Interfaces;
 
 namespace UserService.Commands
 {
     public class LoginUserCommand : ILoginUserCommand
     {
-        IValidator<UserCredentials> validator;
-        UserServiceDbContext dbContext;
-        User user;
-
-        SHA512 shaM = new SHA512Managed();
-
-        public LoginUserCommand(
-            [FromServices] IValidator<UserCredentials> validator,
-            [FromServices] UserServiceDbContext dbContext)
+        public Guid Execute(UserCredentials userCredentials,
+            IValidator<UserCredentials> validator,
+            IUserRepository userRepository)
         {
-            this.validator = validator;
-            this.dbContext = dbContext;
-        }
+            User user;
+
+            SHA512 shaM = new SHA512Managed();
 
-        public Guid Execute(UserCredentials userCredentials)
-        {
             validator.ValidateAndThrow(userCredentials);
 
-            if((user = dbContext.Users.Where(user => user.Email == userCredentials.Email).FirstOrDefault()) == null)
+            if ((user = userRepository.DbContext.Users.Where(
```

Comment:
do not call dbContext from here, create a method GetUser(blabla) in UserRepository and call it here

Reply:
fixed